### ISSUE TITLE
select operator provides incorrect type information when using string paths

### ISSUE THREAD

**Issue by disquisition** [ngrx/platform#940]
## I'm submitting a...

<pre><code>[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request</code></pre>

## What is the current behavior?
Currently, when piping a `select` operator that receives either one string argument or more than two string arguments, the type information is incorrect, resulting in the operator function's return value always being computed as `Store<{}>`. For example:

![screen shot 2018-03-28 at 12 50 32 pm](https://user-images.githubusercontent.com/1706502/38047511-61832aba-3288-11e8-9a67-dc2b6ce521aa.png)

## Expected behavior:
The correct behavior would be for the operator function's return value to be `Store<T>`, where `T` would be replaced by the type of the value at the selected state path. For example:

![screen shot 2018-03-28 at 12 51 21 pm](https://user-images.githubusercontent.com/1706502/38047516-6701c2ee-3288-11e8-8975-4495dde2dce8.png)

## Minimal reproduction of the problem with instructions:
For reference, I've created a [plunker example](https://plnkr.co/edit/zZ9nQRWJ5mfr4j0Z8tw4?p=preview) (check `src/app.ts`), although I don't think you'll be able to see the relevant type information in the browser. You'll likely need to download the files and view them in an editor.

A similar minimal reproduction can be achieved in any existing ngrx project by piping a `select` operator that uses either one or more than two string path segments as its selection method.

## Version of affected browser(s),operating system(s), npm, node and ngrx:
Browser: N/A
OS: macOS 10.12.6
npm: 5.6.0
node: 8.9.3
ngrx: 5.2.0

## Other information:
I have a fix prepared, I'm creating this issue for reference purposes.

**Comment by brandonroberts** [comment]
These changes are already included in #928

**Comment by disquisition** [comment]
@brandonroberts I'm afraid the most important changes from my PR were not present in #928—namely, changing the return types on the relevant `select` overloads from `(source$: Store<a>) => /* ... */` to `(source$: Observable<T>) => /* ... */`. Consequently, I believe this bug will persist.

If you'd like, I can open a new PR. If so, let me know if you'd prefer me to branch off of `master` or `next`.

**Comment by MikeRyanDev** [comment]
@disquisition Yes, if you can create a new PR targeting `next` that would be great! We'll try to get it into the beta we are releasing today.
